### PR TITLE
[BUILD-997] Adds Console Sample for a build that pushes to external registry

### PR DESCRIPTION
**Pull request by apoorvajagtap**
This is an extension to #100.
- Adds a sample example for build pushing to external registry.
- Adds informative comments for references in all the builds. (At present, those comments are ignored when the sample yaml loads while creation of the instance. Checking with console team if this could be corrected with the upcoming versions).

### CODE REVIEW

File: bundle/manifests/openshift-builds-sample-buildah-external-registry.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  yaml: |
+    apiVersion: shipwright.io/v1beta1
+    kind: Build
+    metadata:
+      name: buildah-golang-build-external-push
+    spec:
+      source:
+        type: Git
+        git:
+          url: https://github.com/shipwright-io/sample-go
```
Comment (apoorvajagtap) [inline bundle/manifests/openshift-builds-sample-buildah-external-registry.yaml:21]: Needs reference to rh-openshift-operators sample repo.

**Comment by apoorvajagtap** [comment]
/hold
It'll be better to have the correct `source.url` updated in all the examples with this PR itself.
Ref: https://github.com/redhat-openshift-builds/samples/pull/1

File: bundle/manifests/openshift-builds-sample-buildah-external-registry.yaml (hunk 1 above)
Comment (apoorvajagtap) [reply bundle/manifests/openshift-builds-sample-buildah-external-registry.yaml]: Added the references.

File: bundle/manifests/openshift-builds-sample-buildah-external-registry.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        contextDir: buildah-build
+      strategy:
+        name: buildah
+        kind: ClusterBuildStrategy
+      paramValues:
+      - name: dockerfile
+        value: Dockerfile
+      output:
+        # For a successful image push, a pushSecret with relevant credentials must be available in the same namespace as builds.
+        # Detailed steps are available at: https://docs.openshift.com/builds/1.1/authenticating/understanding-authentication-at-runtime.html#ob-authentication-to-container-registries_understanding-authentication-at-runtime
```
Comment (adambkaplan) [inline bundle/manifests/openshift-builds-sample-buildah-external-registry.yaml:31]: This should point to `docs.redhat.com` - `docs.openshift.com` is being decomissioned soon.
```suggestion
        # Detailed steps are available at: https://docs.redhat.com/en/documentation/builds_for_red_hat_openshift/1.1/html/authentication/understanding-authentication-at-runtime#ob-authentication-to-container-registries_understanding-authentication-at-runtime
```

**Review by adambkaplan — COMMENTED** [review]
Suggestion on URL for the relevant doc, otherwise looks good.

**Review by sayan-biswas — APPROVED** [review]

**Pull request merged**